jsjaws.py

```diff
@@ -344,25 +363,79 @@ def extract_using_soup(self, request: ServiceRequest, file_content):
 
         if aggregated_js_script is None:
-            return None, file_content
+            return None, js_content, None
 
         onloads = soup.body.get_attribute_list("onload")
         for onload in onloads:
             if onload:
-                file_content, aggregated_js_script = self.append_js_content(onload, file_content, aggregated_js_script)
+                js_content, aggregated_js_script = self.append_content(onload, js_content, aggregated_js_script)
 
         aggregated_js_script.close()
 
         request.add_supplementary(aggregated_js_script.name, "temp_javascript.js", "Extracted javascript")
 
-        return aggregated_js_script.name, file_content
+        # Payloads can be hidden in the CSS, so we should try to extract these values and pass them to our JavaScript analysis envs
+        styles = soup.findAll("style")
+        style_json = dict()
+        css_content = b""
+        aggregated_css_script = None
+        for style in styles:
+            # Make sure there is actually a body to the script
+            body = style.string
+            if body is None:
+                continue
+            body = str(body).strip()  # Remove whitespace
+
+            css_content, aggregated_css_script = self.append_content(body, css_content, aggregated_css_script)
+
+            # Parse CSS to JSON
+            qualified_rules = parse_stylesheet(body, skip_comments=True, skip_whitespace=True)
+            for qualified_rule in qualified_rules:
+                preludes = tinycss2_helper.significant_tokens(qualified_rule.prelude)
+                if len(preludes) > 1:
+                    prelude_name = ''.join([prelude.value for prelude in preludes])
+                    self.log.debug(f"Combine all preludes to get the declaration name: {[prelude.value for prelude in preludes]} -> {prelude_name}")
+                else:
+                    prelude_name = preludes[0].value
+                output = tinycss2_helper.parse_declaration_list(qualified_rule.content, skip_comments=True, skip_whitespace=True)
+                style_json[prelude_name] = output
+
+        if aggregated_css_script:
+            aggregated_css_script.close()
+        if aggregated_css_script is None:
+            return aggregated_js_script.name, js_content, None
+
+        if style_json:
+            request.add_supplementary(aggregated_css_script.name, "temp_css.css", "Extracted CSS")
+            css_script_name = aggregated_css_script.name
+
+            # Look for suspicious CSS usage
+            for _, rules in style_json.items():
+                for rule in rules:
+                    declaration_blocks = rule.values()
+                    for declaration_block in declaration_blocks:
+                        for item in declaration_block.get("values", []):
+                            if isinstance(item, dict):
+                                if item.get("url"):
+                                    # SUS
```

Comment:
The sample that we use as an example uses url, but contains a gif. It's not that suspicious. Maybe this should be pushed further and flag actual urls? Are those more suspicious?